src/Common/NamedCollections/NamedCollectionsMetadataStorage.cpp

```diff
@@ -150,28 +150,32 @@ class NamedCollectionsMetadataStorage::LocalStorage : public INamedCollectionsSt
             out.sync();
         out.close();
 
-        fs::rename(tmp_path, getPath(path));
+        fs::rename(tmp_path, getPath(file_name));
     }
 
-    void remove(const std::string & path) override
+    void remove(const std::string & file_name) override
     {
-        if (!removeIfExists(getPath(path)))
+        if (!removeIfExists(file_name))
         {
             throw Exception(
                 ErrorCodes::NAMED_COLLECTION_DOESNT_EXIST,
-                "Cannot remove `{}`, because it doesn't exist", path);
+                "Cannot remove `{}`, because it doesn't exist", file_name);
         }
     }
 
-    bool removeIfExists(const std::string & path) override
+    bool removeIfExists(const std::string & file_name) override
     {
-        return fs::remove(getPath(path));
+        return fs::remove(getPath(file_name));
     }
 
 private:
-    std::string getPath(const std::string & path) const
+    std::string getPath(const std::string & file_name) const
     {
-        return fs::path(root_path) / path;
+        const auto file_name_as_path = fs::path(file_name);
+        if (file_name_as_path.is_absolute())
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Filename {} cannot be an absolute path!", file_name);
+
```

Comment:
I won't write a test because in CI we usually use absolute paths, so it would require a lot of fiddleing. In the other hand @antonio2368 came up with the idea of this extra check. This should be fine and check the issues with absolute paths.

This means:
1. if we use absolute path in config and make a similar mistatke: this check will fail
2. if we don't use absolute in config, then the already existing tests will fail